src/ds/nonblocking.h

```diff
@@ -32,7 +32,7 @@ namespace ringbuffer
 
       PendingMessage(Message m_, std::vector<uint8_t>&& buffer_) :
         m(m_),
-        buffer(buffer_),
+        buffer(std::move(buffer_)),
```

Comment:
this is irrelevant with this PR, just saw it and thought it need an `std::move` here, @eddyashton ?